ui/shared/components/panel/variants/selectors.js

```diff
@@ -73,8 +73,13 @@ export const getPairedSelectedSavedVariants = createSelector(
   (state, props) => (props.project || {}).projectGuid,
   getVariantTagsByGuid,
   getVariantNotesByGuid,
+  state => state.savedVariantsByTag,
   (savedVariants, { tag, gene, familyGuid, analysisGroupGuid, variantGuid }, familiesByGuid, analysisGroupsByGuid,
-    projectGuid, tagsByGuid, notesByGuid) => {
+    projectGuid, tagsByGuid, notesByGuid, savedVariantsByTags) => {
+    if (!projectGuid) {
```

Comment:
For the project page, why not use this if its available? You could skip the `else if (tag && tag !== SHOW_ALL)` block below if you start with this. Also, bailing out early like this totally skips the "paired" part of this "getPairedSelectedSavedVariants" helper for the summary page, which is not an okay change. If you just make the following change and then take out the additional tag filtering below, everything should work fine
```
let variants = savedVariantsByTags[tag] ? savedVariantsByTags[tag].map(guid => savedVariants[guid]) : Object.values(savedVariants)
```

Reply:
Hmm, the project page could also use the cache. But we need to update the saved variant fetching API response handling first. It could fall back to using all loaded variants if the cache is unavailable, but we must keep the summary page from falling back when the cache is missing.

Reply:
BTW. I don't understand what the `paired` means. It looks like pairing the variants with tags and notes. But for what and how?

Reply:
> But we need to update the saved variant fetching API response handling first. It could fall back to using all loaded variants if the cache is unavailable, but we must keep the summary page from falling back when the cache is missing.

It shouldn't be too hard to add a conditional check if that states populated, and then make requiring it conditional on the project

```
let variants = savedVariantsByTags[tag] ? savedVariantsByTags[tag].map(guid => savedVariants[guid]) : Object.values(savedVariants)
if (tag) {
  if (savedVariantsByTags[tag]) {
     variants = savedVariantsByTags[tag].map(guid => savedVariants[guid])
  } else if (!project) {
    return []
  }
}
```

> I don't understand what the paired means

We are displaying compound hets as pairs. If variants are tagged together, they are a comp het pair